### PR TITLE
Add command to list branch deploys

### PR DESCRIPTION
**Pull request by lievendoclo**

### CODE REVIEW

File: lib/java/deploy/MavenPerBranchSpringBootDeploymentGoal.ts [hunk 1]
```diff
@@ -14,9 +14,10 @@
  * limitations under the License.
  */
 
-import { logger } from "@atomist/automation-client";
+import {HandlerContext, logger, Success} from "@atomist/automation-client";
```
Comment (ddgenome) [inline lib/java/deploy/MavenPerBranchSpringBootDeploymentGoal.ts]: Typically we break multiple imports into multiple lines, one for each named import.

File: lib/java/deploy/MavenPerBranchSpringBootDeploymentGoal.ts [hunk 2]
```diff
@@ -34,6 +35,31 @@ import * as os from "os";
 import * as portfinder from "portfinder";
 import { MavenLogInterpreter } from "../../maven/build/mavenLogInterpreter";
 
+export const ListBranchDeploys: CommandHandlerRegistration = {
+    name: "listLocalDeploys",
+    intent: "list branch deploys",
+    description: "list branch deploys",
```
Comment (ddgenome) [inline lib/java/deploy/MavenPerBranchSpringBootDeploymentGoal.ts]: Perhaps something more prose-y and explanatory, e.g., "List deployments of repository across all branches".

File: lib/java/deploy/MavenPerBranchSpringBootDeploymentGoal.ts [hunk 3]
```diff
@@ -34,6 +35,31 @@ import * as os from "os";
...
+    listener: async ci => handleListDeploys(ci.context),
+};
+
+async function handleListDeploysWith(description: string, ctx: HandlerContext) {
```
Comment (ddgenome) [inline lib/java/deploy/MavenPerBranchSpringBootDeploymentGoal.ts]: It does not seem like the `description` is used so it is not clear why you have separated this into two functions.

File: lib/java/deploy/MavenPerBranchSpringBootDeploymentGoal.ts [hunk 4]
```diff
@@ -99,8 +126,10 @@ export function executeMavenPerBranchSpringBootDeploy(projectLoader: ProjectLoad
         try {
             const deployment = await projectLoader.doWithProject({ credentials, id, readOnly: true },
                 project => deployer.deployProject(goalInvocation, project));
-            await goalInvocation.addressChannels(`Deployed \`${id.owner}/${id.repo}/${goalInvocation.sdmGoal.branch} [${
+            const deploymentKey = `${id.owner}/${id.repo}/${goalInvocation.sdmGoal.branch}`;
+            await goalInvocation.addressChannels(`Deployed \`${deploymentKey} [${
                 goalInvocation.sdmGoal.sha}]\` at ${deployment.endpoint}`);
+            deploymentEndpoints[deploymentKey] = {sha: goalInvocation.sdmGoal.sha, endpoint: deployment.endpoint};
```
Comment (ddgenome) [inline lib/java/deploy/MavenPerBranchSpringBootDeploymentGoal.ts]: Typically there are spaces inside curly braces.

File: lib/java/deploy/MavenPerBranchSpringBootDeploymentGoal.ts [hunk 5]
```diff
@@ -99,8 +126,10 @@ export function executeMavenPerBranchSpringBootDeploy(projectLoader: ProjectLoad
         try {
             const deployment = await projectLoader.doWithProject({ credentials, id, readOnly: true },
                 project => deployer.deployProject(goalInvocation, project));
-            await goalInvocation.addressChannels(`Deployed \`${id.owner}/${id.repo}/${goalInvocation.sdmGoal.branch} [${
+            const deploymentKey = `${id.owner}/${id.repo}/${goalInvocation.sdmGoal.branch}`;
+            await goalInvocation.addressChannels(`Deployed \`${deploymentKey} [${
```
Comment (ddgenome) [inline lib/java/deploy/MavenPerBranchSpringBootDeploymentGoal.ts]: Seems odd to break inside a `${}`, although prettier certainly likes to do that.

**Comment by ddgenome** [comment]
Please rebase prior to merge or squash and merge.

File: lib/java/deploy/MavenPerBranchSpringBootDeploymentGoal.ts (hunk 3 above)
Comment (lievendoclo) [reply lib/java/deploy/MavenPerBranchSpringBootDeploymentGoal.ts]: Left over by the `list local deploys` on which this was based. Removed.

File: lib/java/deploy/MavenPerBranchSpringBootDeploymentGoal.ts (hunk 2 above)
Comment (lievendoclo) [reply lib/java/deploy/MavenPerBranchSpringBootDeploymentGoal.ts]: Done

File: lib/java/deploy/MavenPerBranchSpringBootDeploymentGoal.ts (hunk 1 above)
Comment (lievendoclo) [reply lib/java/deploy/MavenPerBranchSpringBootDeploymentGoal.ts]: Done

File: lib/java/deploy/MavenPerBranchSpringBootDeploymentGoal.ts (hunk 5 above)
Comment (lievendoclo) [reply lib/java/deploy/MavenPerBranchSpringBootDeploymentGoal.ts]: Reformatted

File: lib/java/deploy/MavenPerBranchSpringBootDeploymentGoal.ts (hunk 4 above)
Comment (lievendoclo) [reply lib/java/deploy/MavenPerBranchSpringBootDeploymentGoal.ts]: Reformatted

**Pull request merged**